common/desktop-exports

```diff
@@ -103,7 +103,10 @@ mkdir -p $XDG_CONFIG_HOME
 append_dir LOCPATH $RUNTIME/usr/lib/locale
 
 # Run xdg-user-dirs-update
-if [ `which xdg-user-dirs-update` ]; then
+if [ -f $REALHOME/.config/user-dirs.dirs ] && [ -f $REALHOME/.config/user-dirs.locale ]; then
+   ln -s $REALHOME/.config/user-dirs.dirs $HOME/.config/
```

Comment:
This won't work properly, as the XDG_DIRs can be defined as absolute paths or relative to $HOME where $HOME in that case is $REALHOME.

So we can use symlinks to access to such functions and using default local XDG_DIRS (so still generating a local user-dirs.dirs), otherwise and probably better is just to rewrite that file so that it has proper folder paths.

1. Check if file has changed since last time we regenerated (add a .stamp, including what $REALHOME was before, since the user might have changed the location of that)
2. for each XDG_DIR defined, replace $HOME with $REALHOME (expanded variable) so that will be real paths
3. Check if we can access to the defined paths, otherwise fallaback to standard path names inside a snap.